src/lightning/data/streaming/dataset.py

```diff
@@ -85,23 +72,56 @@ def __init__(
         self.index = 0
         self.has_triggered_download = False
         self.min_items_per_replica: Optional[int] = None
-        self.seed = seed
         self.current_epoch = 0
         self.random_state = None
+        self.shuffler: Optional[Shuffle] = None
+
+    def _setup(self) -> None:
+        if self.worker_env is None:
+            self.worker_env = _WorkerEnv.detect()
+
+        env = Environment(dist_env=self.distributed_env, worker_env=self.worker_env)
+
+        # TODO: Why are we conflating input dir and cache dir into one thing? They are conceptually different!
+        # Override the provided input_path
+        cache_dir = _try_create_cache_dir(shard_rank=env.shard_rank)
```

Comment:
We probably need to pass `input_dir` and use it its hash as a sub folder to avoid collision if a collection of StreamingDataset are used together.